src/annotator/integrations/html-metadata.ts

```diff
@@ -47,16 +47,25 @@ export class HTMLMetadata {
    * Returns the primary URI for the document being annotated
    */
   uri(): string {
-    let uri = decodeURIComponent(this._getDocumentHref());
+    let uri = this._getDocumentHref(); // Get the URI without decoding it first
 
-    // Use the `link[rel=canonical]` element's href as the URL if present.
+    // Attempt to decode the URI, handle exceptions if the URI is malformed
+    try {
+      uri = decodeURIComponent(uri);
+    } catch (error) {
+      console.error('Error decoding URI:', error);
+      // Handle the error by logging it and using the original URI
+      uri = this._getDocumentHref();
```

Comment:
The last line should not be needed. If `decodeURIComponent` throws, the value for `uri` will still be what `this._getDocumentHref()` originally returned.

```suggestion
      // Log error for debugging. After this point we fall back to the original URI
      console.error('Error decoding URI:', error);
```